lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,523 @@
+            # user = UserModel.query.filter_by(id=user_id).first()
+
+        except (
+            SQLAlchemyError,
+            DataError,
+            DatabaseError,
+            DBAPIError,
+            CompileError,
+        ) as e:
+            self.logger.error(
+                f"Some SQLAlchemy error occurred while GetUser {e.__dict__['orig']}"
+            )
+            abort(500, message="Some error occurred. Please try again.")
+
+        except Exception as e:
+            self.logger.info(f"Exception occurred during the getUser {e}")
+            abort(500, message="Some exception occurred, please try again")
+
+        # TODO: Check if the user has the right privileges
+
+        if user.username != username:
+            self.logger.info(
+                f"Username retrieved from the auth token {user_id} is different from the username provided"
+            )
+            abort(403, message="Authentication token does not belong to the user")
+
+        response_object = {
+            "status": "success",
+            "data": {
+                "email": user.email,
+                "firstName": user.firstName,
+                "lastName": user.lastName,
+                "registered_on": user.registered_on,
+            },
+        }
+        return make_response(jsonify(response_object), 200)
+
+
+class DeleteUser(Resource):
+    """
+    Abstracted pbench API to delete a user
+    """
+
+    def __init__(self, config, logger, db_session):
+        self.server_config = config
+        self.logger = logger
+        self.db_session = db_session
+
+    def delete(self, username):
+        """
+        Delete request for deleting a user from database.
+        This requires a JWT auth token in the header field
+
+        Required headers include
+
+            Content-Type:   application/json
+            Accept:         application/json
+            X-Auth-Token:   JWT token (user received upon login)
+
+        :return: JSON Payload
+            response_object = {
+                    "status": "success"/"fail",
+                    "status_code": 200/status_code_int,
+                    "message": "Successfully deleted."/"failure message",
+                }
+        """
+        # get the auth token
+        auth_token = get_auth_token(self.logger)
+
+        user_id = UserModel.decode_auth_token(auth_token)
+
+        if isinstance(user_id, str):
+            self.logger.info(
+                f"Token expired, Could not decode the auth token: {auth_token} for user {user_id}"
+            )
+            abort(
+                401, message="Token expired, please log back in",
+            )
+
+        try:
+            # q = UserModel.query.filter_by(id=user_id)
+            q = self.db_session.query(UserModel).filter_by(id=user_id)
+
+            # TODO: Check if the user has the right privileges
+
+            if q.first().username != username:
```

Comment:
We probably want "admin" to be able to delete any user, so using your privilege check would probably be good.

We might want to move towards a CRUD priv model, where you ask instead "can I perform this operation on this resource" where "can I delete my own user" would be True, "can I delete someone else" would be False unless "I" am admin... doesn't have to be a first step, though!